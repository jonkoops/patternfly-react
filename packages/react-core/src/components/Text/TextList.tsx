import { HTMLProps, ReactNode, FunctionComponent } from 'react';
import styles from '@patternfly/react-styles/css/components/Content/content';
import { css } from '@patternfly/react-styles';

export enum TextListVariants {
  ul = 'ul',
  ol = 'ol',
  dl = 'dl'
}

export interface TextListProps extends HTMLProps<HTMLElement> {
  /** Content rendered within the TextList */
  children?: ReactNode;
  /** Additional classes added to the TextList */
  className?: string;
  /** The text list component */
  component?: 'ul' | 'ol' | 'dl';
  /** Modifies the list to include plain styling */
  isPlain?: boolean;
}

export const TextList: FunctionComponent<TextListProps> = ({
  children = null,
  className = '',
  component = TextListVariants.ul,
  isPlain = false,
  ...props
}: TextListProps) => {
  const Component: any = component;

  return (
    <Component {...props} className={css(isPlain && styles.modifiers.plain, className)}>
      {children}
    </Component>
  );
};
TextList.displayName = 'TextList';
