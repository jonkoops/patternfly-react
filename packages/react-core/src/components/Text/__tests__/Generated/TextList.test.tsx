/**
 * @vitest-environment jsdom
 */
import { render } from '@testing-library/react';
import React from 'react';
import { expect, it } from 'vitest';

import { TextList } from '../../TextList';

it('TextList should match snapshot (auto-generated)', () => {
  const { asFragment } = render(<TextList children={<>ReactNode</>} className={"''"} component={'ul'} />);
  expect(asFragment()).toMatchSnapshot();
});
