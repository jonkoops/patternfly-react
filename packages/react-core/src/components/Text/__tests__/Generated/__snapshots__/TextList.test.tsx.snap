// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextList should match snapshot (auto-generated) 1`] = `
<DocumentFragment>
  <ul
    class="''"
  >
    ReactNode
  </ul>
</DocumentFragment>
`;
